torch/distributed/_tensor/device_mesh.py

```diff
@@ -152,6 +153,11 @@ def _get_or_create_default_group(self):
                 # automatically set the current cuda device base on num of gpu devices available in each host
                 # NOTE: This device selection would only work for homogenous hardware.
                 torch.cuda.set_device(get_rank() % torch.cuda.device_count())
+            # to perform DTensor random ops, we need to ensure all ranks in mesh is initialized with the same
+            # random seed. The seed to use will be the current seed on rank 0.
+            object_list = [torch.cuda.initial_seed()]
+            broadcast_object_list(object_list)
+            torch.cuda.manual_seed(int(object_list[0]))
```

Comment:
I am not sure if we should by default manual_seed here in the device mesh constructor(doing it in `manual_seed` call is ok), we should probably first assign the initial seed as an attribute of device_mesh. And we need to think about how to use it.

Reply:
sounds good.